### PR TITLE
Rework the output website elements and drop python 2

### LINKED ISSUE #19: Better CSS and JS
Currently, some of our Javascript is writing in-place CSS styling for some HTML elements. Sometimes this is due to desired conditional handling of styling, but other times, it's done as a convenience for a specific problem. 

Overall, it'd be better to use classes/ids for these types of issues. Also, allowing an extension to specify an extra CSS style sheet (or sheets) and providing utilities to adjust the livvkit element html classes and ids would cover most use cases.

### LINKED ISSUE #20: Refactor LIVVkit elements
Jupyter notebooks can render custom html if a class implements the `_repr_html` method. If the element definitions were refactored as objects they could include `_repr_html` methods which would just call the relevant javascript functions inside a script tag.  

If this was done I think the TeX helper module could be merged into the main element subclasses as well.

@jhkennedy I just randomly had this idea after learning about `_repr_html` and figured I'd throw it out there.  Feel free to close if you don't think it's a good/feasible addition.

### LINKED ISSUE #30: Drop Python 2 support
[Python 2 End Of Life has been declared; it will die in 2020.](https://pythonclock.org/)

We should not support python 2 after PyCon 2020.

---

*Le roi est mort, vive le roi!*

### LINKED ISSUE #31: Add element for V-H Table
Currently, there is a `js` method to render a table with both vertical and horizontal headers (`V-H Table`), but there is no associated LIVVkit element.

### PR DESCRIPTION
**Pull request by jhkennedy**
currently, to produce the website as viewed in the browser, LIVVkit relies on a collection of Javascript functions that parse associated JSON files to dynamically create HTML. This is overly complicated as LIVVkit websites are static and makes customizing LIVVkit display elements hard (you have to edit the Javascript well away from where you create elements). 

This PR reworks the elements to be produced from customizable HTML templates through jinja2, and significantly simplifies both working with the elements and the required Javascript for the website. 

Additionally, as part of this effort, working with the classes and templates is significantly easier without trying to support python 2. Because Python 2 EOL is already here, this also drops python 2 support, moving the supported and tested python versions to 3.6 and up. 

Because of the significance of these changes, develop should now be considered to target the LIVVkit 3.0.0 release (version number bumped accordingly) and this major API change will break associated packages like LEX and EVV without updates. 

Fixes #30 
Fixes #19 
Fixes #20 
Closes #31 (obsolete)

### CODE REVIEW

File: livvkit/bundles/CISM_glissade/verification.py [hunk 1]
```diff
@@ -52,14 +50,9 @@ def parse_log(file_path):
         the results of the bit for bit testing
     """
     if not os.path.isfile(file_path):
-        return elements.error("Output Log", "Could not open file: " + file_path.split(os.sep)[-1])
+        return elements.Error("Output Log", "Could not open file: " + file_path.split(os.sep)[-1]).__dict__
```
Comment (mkstratos) [inline livvkit/bundles/CISM_glissade/verification.py]: This returns a dict rather than the `elements.Error` Element, is that correct?

**Review by mkstratos — CHANGES_REQUESTED** [review]
Aside from the one (possible) change in `livvkit/bundles/CISM_glissade/verification.py`, all looks good to me. Though, does `tests\conftest.py` ever get called when testing?

File: livvkit/bundles/CISM_glissade/verification.py (hunk 1 above)
Comment (jhkennedy) [reply livvkit/bundles/CISM_glissade/verification.py]: Ooh, good catch.

**Review by mkstratos — APPROVED** [review]

**Comment by jhkennedy** [comment]
@mkstratos I updated docs to reflect all these changes (and the changes in dev too). Once you're happy I'll do the merge. 

as for `conftest.py`, that's used by `pytest` when it's run -- it sets up some shared methods that will be available to all the tests. See:
* https://docs.pytest.org/en/2.7.3/plugins.html?highlight=re
* https://stackoverflow.com/questions/34466027/in-pytest-what-is-the-use-of-conftest-py-files

**Review by mkstratos — APPROVED** [review]
Joe, all looks good to me. Can you tag any other issues this will close?

**Comment by jhkennedy** [comment]
Alright, all issues tagged and closed that this one fixes, plus I added some comments on other issues. 

Couple questions: 

1. Is this what we want to release for LIVVkit 3.0? 
2. If so, do you want:
    1. Me to do all the release steps? (merge to master, pypi, conda-forge)
    2. You to do all the release step?
   
   Either way, I'll write add documentation for how to do it all

**Pull request merged**